internal/host/hostcommands/domain_name_resolution_cmd.go

```diff
@@ -0,0 +1,96 @@
+package hostcommands
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/jinzhu/gorm"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/models"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type domainNameResolutionCmd struct {
+	baseCmd
+	domainNameResolutionImage string
+	db                        *gorm.DB
+}
+
+func NewdomainNameResolutionCmd(log logrus.FieldLogger, domainNameResolutionImage string, db *gorm.DB) *domainNameResolutionCmd {
+	return &domainNameResolutionCmd{
+		baseCmd:                   baseCmd{log: log},
+		domainNameResolutionImage: domainNameResolutionImage,
+		db:                        db,
+	}
+}
+
+func (f *domainNameResolutionCmd) prepareParam(host *models.Host, cluster *common.Cluster) (string, error) {
+	clusterName := cluster.Cluster.Name
+	if len(clusterName) == 0 {
+		err := errors.Errorf("Cluster name is empty for cluster %s", host.ClusterID)
+		f.log.WithError(err).Warn("Cluster name is empty")
+		return "", err
+	}
+
+	baseDNSDomain := cluster.Cluster.BaseDNSDomain
+	if len(baseDNSDomain) == 0 {
+		err := errors.Errorf("Cluster base domain is empty for cluster %s", host.ClusterID)
+		f.log.WithError(err).Warn("Cluster base domain is empty")
+		return "", err
+	}
+	apiDomainName := fmt.Sprintf("api.%s.%s", clusterName, baseDNSDomain)
+	apiInternalDomainName := fmt.Sprintf("api-int.%s.%s", clusterName, baseDNSDomain)
+	appsDomainName := fmt.Sprintf("console-openshift-console.apps.%s.%s", clusterName, baseDNSDomain)
```

Comment:
Can we place `console-openshift-console` in some global constant and use that constant in all 3 places where it is used? The constant name should not be specific to the console. If we ever decide to change the subdomain that we check, I was it to be as easy as changing one line.

Also, above that constant's definition please write a comment describing that this is an arbitrary subdomain of `*.apps.<cluster>.<base>` that we use in order to validate that the wildcard DNS has been applied by the user correctly during DNS validations

Reply:
Changed, create new const under constant, and replace all usages.